src/server/frontend_wayland/generator/interface.cpp

```diff
@@ -183,6 +233,39 @@ Emitter Interface::member_vars() const
     }
 }
 
+Emitter Interface::enum_declarations() const
+{
+    std::vector<Emitter> ret;
+    for (auto const& i : enums)
+    {
+        ret.push_back(i.declaration());
+    }
+    return List{ret, empty_line};
+}
+
+Emitter Interface::event_opcodes() const
+{
+    std::vector<Emitter> ret;
+    for (auto const& i : events)
+    {
+        ret.push_back(i.opcode_declare());
+    }
+    Emitter body = List{ret, nullptr};
+    if (body.is_valid())
+    {
+        return Lines{
+            {"struct Opcode"},
```

Comment:
Again, this seems like `enum class` would be a better translation.

Reply:
I did it this way for consistency with enums.